idea/idea-fir/src/org/jetbrains/kotlin/idea/quickfix/fixes/InitializePropertyQuickFixFactories.kt

```diff
@@ -38,8 +38,15 @@ object InitializePropertyQuickFixFactory {
 
     @OptIn(ExperimentalStdlibApi::class)
     val initializePropertyFactory =
-        diagnosticFixFactory(KtFirDiagnostic.MustBeInitializedOrBeAbstract::class) { diagnostic ->
+        diagnosticFixFactories(
+            KtFirDiagnostic.MustBeInitialized::class,
+            KtFirDiagnostic.MustBeInitializedOrBeAbstract::class
+        ) { diagnostic ->
             val property: KtProperty = diagnostic.psi
+
+            // An extension property cannot be initialized because it has no backing field
```

Comment:
Can this be more general? E.g., by using `KtPropertySymbol.hasBackingField`?

Reply:
It seems to be returning `true` for the case where I had to add this:
```kt
class A {
    <caret>var Int.n: Int
        get() = 1
}
```

`hasBackingField` only returns `false` once the setter is added. Looking it its implementation here and in FE 1.0, neither of them explicitly checks if it's an extension property, and that doesn't seem to an issue. I tried to add an explicit check to see if it's an extension property, however this caused a lot of errors. E.g., [this entire block in `FirDeclarationCheckerUtils.kt`](https://github.com/JetBrains/kotlin/blob/84a7bdffe5217981d90a688fbd485b811d4635c2/compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/declaration/FirDeclarationCheckerUtils.kt#L148-L158) assumes `hasBackingField` return `true` even for extension properties (also L124-L129). I would prefer not to address all that in this PR.

That said, I don't think it is correct to report `MUST_BE_INITIALIZED` _ever_ for extension properties. In the example above, `MUST_BE_INITIALIZED` is reported in FE 1.0 and FIR. But I think the error should indicate that it should have both accessors. Maybe `EXTENSION_PROPERTY_MUST_HAVE_ACCESSORS_OR_BE_ABSTRACT` instead. But let's leave that off this PR for now.

Reply:
Note that the code block you mentioned in FIR part is quite identical to the counterpart in FE1.0: https://github.com/JetBrains/kotlin/blob/master/compiler/frontend/src/org/jetbrains/kotlin/resolve/DeclarationsChecker.kt#L735-L744 So, if we want to introduce a new diagnostic, maybe need to do so for both ends.

Reply:
Yes, I agree. The add accessors quick fix could also be used for `EXTENSION_PROPERTY_MUST_HAVE_ACCESSORS_OR_BE_ABSTRACT`.